presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -935,5 +942,16 @@ public synchronized void generateCpuQuota(long elapsedSeconds)
                 internalGenerateCpuQuota(elapsedSeconds);
             }
         }
+
+        public void setTaskLimitExceeded(boolean exceeded)
+        {
+            this.taskLimitExceeded.set(exceeded);
+        }
+
+        private boolean isResourceLimitExceeded()
+        {
+            // can be expanded to include other metrics
+            return taskLimitExceeded.get();
+        }
```

Comment:
If we follow my suggestion above, we can avoid this call. I will say that the plumbing is not worth it and we should just implement it when we have a concrete use case.